### PR TITLE
LPS-131886 Test Auto WC Exhibition in Content Dashboard

### PR DESCRIPTION
**Pull request by Luizcarlosqueiroz**
**JIRA Ticket:**
LPS-131886
LPS-111322

**Motivation:**
Test automation for the story "As a content author, I want to see all my content listed in a single view"

### CODE REVIEW

**Comment by Luizcarlosqueiroz** [comment]
Changed initial tests name according to [LPS-130691](https://issues.liferay.com/browse/LPS-130691).

from: ContentDashboardExhibitWebContentAssociatedDisplayPageTemplate
to:  ShouldExhibitWebContentAssociatedWithDisplayPage

from: ContentDashboardExhibitWebContent
to: ShouldExhibitWebContentNotAssociatedWithDisplayPageTemplate 

from: ContentDashboardExhibitEditedWebContent
to: ShouldExhibitUpdatedWebContent

**Comment by Luizcarlosqueiroz** [comment]
@CarlosBrichete

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 1]
```diff
@@ -308,110 +308,128 @@ definition {
...
+			ApplicationsMenu.gotoPortlet(
+				category = "Content",
+				panel = "Applications",
+				portlet = "Content Dashboard");
 
-		ApplicationsMenu.gotoPortlet(
-			category = "Content",
-			panel = "Applications",
-			portlet = "Content Dashboard");
+			AssertElementNotPresent(locator1 = "//span[contains(@title, 'This content has a display page')]");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: You cannot put a path within an assert. You have to create the path element for the Display Page icon in Content dashboard.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 2]
```diff
@@ -308,110 +308,128 @@ definition {
...
-			portlet = "Content Dashboard");
+			WebContent.viewTitle(webContentTitle = "WC WebContent New Title");
 
-		AssertTextPresent(
-			locator1 = "ContentDashboard#FIRST_ELEMENT_LISTED",
-			value1 = "WC WebContent New Title");
+			WebContent.deleteCP(
+				siteURLKey = "test-site-name",
+				webContentTitle = "WC WebContent New Title");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: No need to delete the WC, you will delete the site in the teardown()

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase (hunk 1 above)
Comment (CarlosBrichete) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: You should validate that the Display Page associated icon is not present for the WC created. A way to do this is validate that just the WC is listed in the table and the Display Page associated icon is not present.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 3]
```diff
@@ -308,110 +308,128 @@ definition {
...
 	@priority = "5"
-	test ContentDashboardExhibitWebContentAssociatedDisplayPageTemplate {
-		JSONWebcontent.addWebContent(
-			content = "WC WebContent Content",
-			groupName = "Test Site Name",
-			title = "WebContent Title");
+	test ShouldExhibitWebContentAssociatedWithDisplayPage {
+		task ("Add a site via JSON") {
+			JSONGroup.addGroup(groupName = "Test Site Name");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: Why you create another Test Site Name? You are already creating it in the setUp

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 4]
```diff
@@ -308,110 +308,128 @@ definition {
...
+			PortletEntry.publish();
+		}
+
+		task ("Check in Content Dashboard if WC is assiciated with Display Page Template") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Content",
+				panel = "Applications",
+				portlet = "Content Dashboard");
+			
+			AssertElementPresent(locator1 = "//span[contains(@title, 'This content has a display page')]");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: You cannot put a path within an assert. You have to create the path element for the Display Page icon in Content dashboard.

Comment (CarlosBrichete) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: You should validate that the Display Page associated icon is not present for the WC created. A way to do this is validate that just the WC is listed in the table and the Display Page associated icon is not present.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 5]
```diff
@@ -308,110 +308,128 @@ definition {
...
-			value1 = "WC Title");
+		task ("Delete WC") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = "test-site-name");
+
+			WebContent.viewTitle(webContentTitle = "WC Title");
+
+			WebContent.deleteCP(
+				siteURLKey = "test-site-name",
+				webContentTitle = "WC Title");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: No need to delete the WC, you will delete the site in the teardown()

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 6]
```diff
@@ -308,110 +308,128 @@ definition {
...
+
+		task ("Delete WC") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = "test-site-name");
+
+			WebContent.viewTitle(webContentTitle = "Webcontent Title");
+
+			WebContent.deleteCP(
+				siteURLKey = "test-site-name",
+				webContentTitle = "Webcontent Title");
+		}
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: No need to delete the WC, you will delete the site in the teardown()

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 7]
```diff
@@ -308,110 +308,128 @@ definition {
...
-			panel = "Applications",
-			portlet = "Content Dashboard");
+			WebContent.editDisplayPage(
+				pageName = "Display Page Name",
+				pageType = "Display Page Templates");
 
-		AssertTextPresent(
-			locator1 = "ContentDashboard#FIRST_ELEMENT_LISTED",
-			value1 = "WebContent Title");
+			PortletEntry.publish();
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: There's no need to edit the WC. It is easier if you just set the Display Page you have created before as Default

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase (hunk 3 above)
Comment (Luizcarlosqueiroz) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: Totally, got confused form some reason. I removed this now.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase (hunk 4 above)
Comment (Luizcarlosqueiroz) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase]: In this case, we want to assert that the icon is being presented.

**Comment by Luizcarlosqueiroz** [comment]
@CarlosBrichete I fixed the tests according to review.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase [hunk 8]
```diff
@@ -1101,17 +1085,7 @@ definition {
...
-
-		task ("Delete WC") {
-			WebContentNavigator.openWebContentAdmin(siteURLKey = "test-site-name");
-
-			WebContent.viewTitle(webContentTitle = "Webcontent Title");
-
-			WebContent.deleteCP(
-				siteURLKey = "test-site-name",
-				webContentTitle = "Webcontent Title");
+			AssertElementPresent(locator1 = "ContentDashboard#FIRST_ELEMENT_LISTED_DISPLAY_PAGE");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentdashboard/ContentDashboard.testcase:1088]: In the test _ShouldExhibitWebContentNotAssociatedWithDisplayPageTemplate_ , you assert both the WC name is present and the Display Page icon is not present, but in this test you are just asserting the Display Page icon present. Why don't you assert the WC name in this test also?

**Comment by Luizcarlosqueiroz** [comment]
@CarlosBrichete

**Comment by CarlosBrichete** [comment]
ci:test:sf

**Comment by CarlosBrichete** [comment]
ci:test:tango

**Comment by CarlosBrichete** [comment]
ci:forward

**Pull request closed without merging**